Classes/TextUtils.cs

```diff
@@ -32,6 +33,7 @@ public static class TextUtils
         private const string MatchMonthText = @"(?:Jan(?:uary)?|Feb(?:ruary)?|Mar(?:ch)?|Apr(?:il)?|May|Jun(?:e)?|Jul(?:y)?|Aug(?:ust)?|Sep(?:t(?:ember)?)?|Oct(?:ober)?|Nov(?:ember)?|Dec(?:ember)?)";
         private const string MatchYear = @"(?:(?:\d{4})|(?:\d{2})(?![\\d])|(?:\')(?:\d{2})(?![\\d]))";
         private const string MatchWS = @"(?: ?)";
+        private const string MatchWSDelimEnd = @"((?:\.|\-|\/| |:)(?=(?:\.|\-|\/| |:)|$))|(\s(?:\.|\-|\/| |:)(?=\s(?:\.|\-|\/| |:)|$))";
```

Comment:
Is there a reason for the complexity of this regex, or would the following suffice:

    private const string MatchWSDelimEnd = @"[.\/ :-]+$";

Reply:
At the time I needed to be able to optionally handle leading spaces, and that complex regex was what i able to get working. Will try out your recommendation and see if the tests still pass.